src/Swift.Bindings/src/Parser/SwiftABIParser.cs

```diff
@@ -146,7 +146,7 @@ public ModuleDecl GetModuleDecl()
 
             moduleDecl.Fields = decls.OfType<FieldDecl>().ToList();
             moduleDecl.Methods = decls.OfType<MethodDecl>().ToList();
-            moduleDecl.Declarations = decls.Where(d => !(d is MethodDecl) && !(d is FieldDecl)).ToList();
+            moduleDecl.Types = decls.OfType<TypeDecl>().ToDictionary(d => new NamedTypeSpec(d.FullyQualifiedName), d => d);
```

Comment:
One thing that we will run into right here is that the `*Decl` types are already C# and that type may not be the same as the Swift from which it came. For example, if we get a Swift type that is unpronounceable in C# (eg, contains emoji) then we've just created an incorrect NamedTypesSpec.